# Patch is fixing: https://github.com/tvheadend/tvheadend/issues/1833
diff -urN ../ffmpeg-7.1.3.orig/libavcodec/hw_base_encode.c ./libavcodec/hw_base_encode.c
--- ../ffmpeg-7.1.3.orig/libavcodec/hw_base_encode.c    2023-04-03 00:53:26.000000000 +0200
+++ ./libavcodec/hw_base_encode.c    2023-04-03 07:46:36.641867975 +0200
@@ -504,7 +504,7 @@ static int hw_base_encode_send_frame(AVCodecContext *avctx, FFHWBaseEncodeContex
 
         // Fix timestamps if we hit end-of-stream before the initial decode
         // delay has elapsed.
-        if (ctx->input_order <= ctx->decode_delay)
+        if (ctx->input_order <= ctx->decode_delay && ctx->pic_end)
             ctx->dts_pts_diff = ctx->pic_end->pts - ctx->first_pts;
     }
 
