diff -urN ../x265_2.0.pic/source/CMakeLists.txt ./source/CMakeLists.txt
--- ../x265_2.0.pic/source/CMakeLists.txt	2016-07-30 08:57:07.632539944 +0200
+++ ./source/CMakeLists.txt	2016-08-08 12:12:39.539380667 +0200
@@ -168,7 +168,7 @@
     add_definitions(-D__STDC_LIMIT_MACROS=1)
     add_definitions(-std=gnu++98)
     if(ENABLE_PIC)
-         add_definitions(-fPIC)
+         add_definitions(-fPIE)
     endif(ENABLE_PIC)
     if(NATIVE_BUILD)
         if(INTEL_CXX)
@@ -183,11 +183,11 @@
         endif()
     endif()
     if(ARM AND CROSS_COMPILE_ARM)
-        set(ARM_ARGS -march=armv6 -mfloat-abi=soft -mfpu=vfp -marm -fPIC)
+        set(ARM_ARGS -march=armv6 -mfloat-abi=soft -mfpu=vfp -marm -fPIE)
     elseif(ARM)
         find_package(Neon)
         if(CPU_HAS_NEON)
-            set(ARM_ARGS -mcpu=native -mfloat-abi=hard -mfpu=neon -marm -fPIC)
+            set(ARM_ARGS -mcpu=native -mfloat-abi=hard -mfpu=neon -marm -fPIE)
             add_definitions(-DHAVE_NEON)
         else()
             set(ARM_ARGS -mcpu=native -mfloat-abi=hard -mfpu=vfp -marm)
