--- source/CMakeLists.txt.old	2016-05-03 14:34:37.168396127 +0200
+++ source/CMakeLists.txt	2016-05-03 14:35:35.551372285 +0200
@@ -100,7 +100,7 @@
     endif(NO_ATOMICS)
 endif(UNIX)
 
-if(X64 AND NOT WIN32)
+if(X64NONONO AND NOT WIN32)
     option(ENABLE_PIC "Enable Position Independent Code" ON)
 else()
     option(ENABLE_PIC "Enable Position Independent Code" OFF)
@@ -166,6 +166,8 @@
     add_definitions(-std=gnu++98)
     if(ENABLE_PIC)
          add_definitions(-fPIC)
+    else()
+         add_definitions(-fPIE)
     endif(ENABLE_PIC)
     if(NATIVE_BUILD)
         if(INTEL_CXX)
--- source/cmake/CMakeASM_YASMInformation.cmake.old	2016-05-03 15:52:36.572122457 +0200
+++ source/cmake/CMakeASM_YASMInformation.cmake	2016-05-03 15:53:18.939438179 +0200
@@ -3,9 +3,7 @@
 
 if(X64)
     list(APPEND ASM_FLAGS -DARCH_X86_64=1)
-    if(ENABLE_PIC)
-        list(APPEND ASM_FLAGS -DPIC)
-    endif()
+    list(APPEND ASM_FLAGS -DPIC)
     if(APPLE)
         set(ARGS -f macho64 -m amd64 -DPREFIX)
     elseif(UNIX AND NOT CYGWIN)
