diff -urN ../x265_2.0.orig/source/CMakeLists.txt ./source/CMakeLists.txt
--- ../x265_2.0.orig/source/CMakeLists.txt	2016-07-13 15:53:26.000000000 +0200
+++ ./source/CMakeLists.txt	2016-07-30 08:57:07.632539944 +0200
@@ -107,11 +107,7 @@
     endif(NO_ATOMICS)
 endif(UNIX)
 
-if(X64 AND NOT WIN32)
-    option(ENABLE_PIC "Enable Position Independent Code" ON)
-else()
-    option(ENABLE_PIC "Enable Position Independent Code" OFF)
-endif(X64 AND NOT WIN32)
+option(ENABLE_PIC "Enable Position Independent Code" ON)
 
 # Compiler detection
 if(CMAKE_GENERATOR STREQUAL "Xcode")
