--- capmt.c	2013-10-11 21:22:04.000000000 +0800
+++ capmt.c	2013-10-11 16:45:25.000000000 +0800
@@ -826,10 +826,13 @@
                 cad.cad_length = 0x07;
                 cad.cad_data[5] = cce2->cce_providerid >> 8;
                 cad.cad_data[6] = cce2->cce_providerid & 0xffffff;
-              } else if (cce2->cce_caid >> 8 == 0x4a) {
+              } else if (cce2->cce_caid >> 8 == 0x4a && cce2->cce_caid!=0x4ad2) {
                 cad.cad_length = 0x05;
                 cad.cad_data[4] = cce2->cce_providerid & 0xffffff;
-              } else
+              }else if (cce2->cce_caid == 0x4ad2) {//streamguard
+                  cad.cad_length = 0x04;
+                  cad.cad_data[3] = cce2->cce_providerid & 0xffffff;
+              }else
                 tvhlog(LOG_WARNING, "capmt", "Unknown CAID type, don't know where to put provider ID");
             }
             memcpy(&buf[pos], &cad, cad.cad_length + 2);
