--- psi.c.bak	2013-10-11 21:18:31.000000000 +0800
+++ psi.c	2013-10-11 21:53:40.000000000 +0800
@@ -270,11 +270,15 @@
       i += nanolen;
     }
     break;
-  case 0x4a00://DRECrypt
-    if (caid != 0x4aee) { // Bulcrypt
+  case 0x4a00://DRECrypt or streamguard
+    if (caid==0x4ad2) {
+              provid=0;
+    }
+    if (caid != 0x4aee && caid!=0x4ad2) { // Bulcrypt
       provid = size < 4 ? 0 : buffer[4];
       break;
     }
+          
   default:
     provid = 0;
     break;
@@ -489,11 +493,13 @@
     case 0x04:
       hts_stream_type = SCT_MPEG2AUDIO;
       break;
-
+            
     case 0x81:
       hts_stream_type = SCT_AC3;
       break;
-    
+    case 0x06:
+            hts_stream_type = SCT_AC3;
+             break;
     case 0x0f:
       hts_stream_type = SCT_MP4A;
       break;
@@ -898,6 +904,7 @@
   { "Bulcrypt",         0x4aee },
   { "Bulcrypt",         0x5581 },
   { "Verimatrix",       0x5601 },
+  { "StreamGuard",      0x4ad2 },
 };
 
 const char *
